### PR TITLE
Use 20% user input data as testing data

### PR DESCRIPTION
**Pull request by abs51295**
- This PR introduces the calculation of precision and recall on user input data.
- Training is only made availabe through manifest.json from now on.

### CODE REVIEW

File: training/train.py [hunk 1]
```diff
@@ -127,54 +116,46 @@ def preprocess_raw_data(raw_data_dict, lower_limit, upper_limit):
...
+    df_user = user_input_df.drop_duplicates(['UserId'])
+    user_input_df = user_input_df.sample(frac=1)
+    df_item = user_input_df.drop_duplicates(['ItemId'])
     train_df = pd.concat([df_user, df_item]).drop_duplicates()
-    fraction = round(frac(data_df, train_df), 2)
+    fraction = round(frac(user_input_df, train_df), 2)
 
     if fraction < 0.80:
-        df_ = extra_df(fraction, data_df, train_df)
+        df_ = extra_df(fraction, user_input_df, train_df)
```
Comment (lucky-suman) [inline training/train.py]: Are we not taking the data in which 'is_user_input_stack' is False?

Comment (abs51295) [reply training/train.py]: We do that in line 156 and 157. This is for 80-20 split of user input stack only.

File: training/train.py [hunk 2]
```diff
@@ -127,54 +116,46 @@ def preprocess_raw_data(raw_data_dict, lower_limit, upper_limit):
...
         train_df = pd.concat([train_df, df_])
-    test_df = pd.concat([data_df, train_df]).drop_duplicates(keep=False)
+    test_df = pd.concat([user_input_df, train_df]).drop_duplicates(keep=False)
+    test_df = test_df.drop(columns=['is_user_input_stack'])
+    data_df = data_df.loc[~data_df['is_user_input_stack']]
+    train_df = pd.concat([data_df, train_df])
```
Comment (lucky-suman) [inline training/train.py]: Help me understand if training DataFrame has reached 80% of given data, then why we are adding more data into it?

Comment (abs51295) [reply training/train.py]: It's only 80% of user input data. What about bigquery data? We need to add the BQ data as well and hence we are adding 100% of BQ data. Hence total data is now 180% of total data.

File: training/train.py (hunk 1 above)
Comment (lucky-suman) [reply training/train.py]: Ohk, yeah I got it. You are taking both data separately.

File: training/train.py (hunk 2 above)
Comment (lucky-suman) [reply training/train.py]: Understood thanks :)

**Pull request merged**